tests/helpers.test.ts

```diff
@@ -1,21 +1,68 @@
 jest.unmock('../src/helpers')
 jest.mock('fs')
-
-import { pathToJestPackageJSON, isCRATestCommand } from '../src/helpers'
-import { existsSync } from 'fs'
+jest.mock('vscode', () => ({
+  window: {
+    showWarningMessage: () => {},
+  },
+}))
+
+import { pathToJestPackageJSON, isCRATestCommand, pathToJest } from '../src/helpers'
+import { existsSync, readFileSync } from 'fs'
 import * as path from 'path'
 
 const existsMock = existsSync as jest.Mock<boolean>
-const defaultPathToJest = 'node_modules/.bin/jest'
+const readFileMock = readFileSync as jest.Mock<string>
 
 describe('ModuleHelpers', () => {
+  describe('pathToJest', () => {
+    it('should return the set value', () => {
+      const workspace: any = {
+        pathToJest: 'abcd',
+      }
+      expect(pathToJest(workspace)).toBe(workspace.pathToJest)
+    })
+
+    it('should recognize CRA apps', () => {
+      readFileMock.mockReturnValueOnce('{"scripts":{"test":"react-scripts test"}}')
+
+      const workspace: any = {
+        rootPath: '',
+        pathToJest: '',
+      }
+      expect(pathToJest(workspace)).toBe('npm test --')
+    })
+
+    it('should default to `node_modules/.bin/jest`', () => {
+      readFileMock.mockReturnValueOnce('{}')
+      existsMock.mockReturnValueOnce(true)
+
+      const defaultPath = path.normalize('node_modules/.bin/jest')
+      const workspace: any = {
+        rootPath: '',
+        pathToJest: '',
+      }
+      expect(pathToJest(workspace)).toBe(defaultPath)
+    })
+
+    it('should fallback to `jest` if everything other fails', () => {
+      readFileMock.mockReturnValueOnce('{}')
+      existsMock.mockReturnValueOnce(false)
+
+      const workspace: any = {
+        rootPath: '',
+        pathToJest: '',
+      }
+      expect(pathToJest(workspace)).toBe('jest')
+    })
+  })
+
```

Comment:
Nice. This makes me sad we duplicated this effort. 😭

Reply:
At least we now can keep the best of both PRs 😉. Sorry that I didn't mention it already when discussing [324](https://github.com/jest-community/vscode-jest/pull/324#issuecomment-390658165).